### PR TITLE
Remove `torch/csrc/generic`

### PR DESCRIPTION
**Pull request by kurtamohler**
### Description
Remove `torch/csrc/generic` since it is no longer needed.

### Issue
#82372

### Testing
No tests added

### CODE REVIEW

File: torch/csrc/generic/utils.h [hunk 1]
```diff
@@ -1,33 +0,0 @@
...
-#else
-
-#if defined(TH_REAL_IS_HALF)
-#define GENERATE_SPARSE 0
-#else
-#define GENERATE_SPARSE 1
-#endif
-
-struct THPStorage;
-struct THSPTensor;
```
Comment (kurtamohler) [inline torch/csrc/generic/utils.h:12]: I think this `THSPTensor` thing is no longer used at all, and there are other files related to it that could be cleaned up. I'll find out for sure and remove that stuff in another PR if so.

**Review by ezyang — APPROVED** [review]

**Comment by kurtamohler** [comment]
`pull / win-vs2019-cpu-py3 / test (default, 1, 2, windows.4xlarge)` failure appears to be upstream

**Comment by kurtamohler** [comment]
Guess I need to rebase

**Pull request closed without merging**